winch/codegen/src/codegen/context.rs

```diff
@@ -231,6 +231,27 @@ impl<'a, 'builtins> CodeGenContext<'a, 'builtins> {
         self.stack.push(dst.into());
     }
 
+    /// Prepares arguments for emitting an f32 or f64 comparison operation.
+    pub fn float_cmp_op<F, M>(&mut self, masm: &mut M, size: OperandSize, mut emit: F)
+    where
+        F: FnMut(&mut M, Reg, Reg, Reg, OperandSize),
+        M: MacroAssembler,
+    {
+        let src1 = self.pop_to_reg(masm, None);
+        let src2 = self.pop_to_reg(masm, None);
+        let dst = self.any_gpr(masm);
+        emit(masm, dst, src1.reg, src2.reg, size);
+        self.free_reg(src1);
+        self.free_reg(src2);
+
+        let ty = match size {
+            OperandSize::S32 => WasmType::I32,
+            OperandSize::S64 => WasmType::I64,
+            OperandSize::S128 => unreachable!(),
+        };
+        self.stack.push(TypedReg::new(ty, dst).into());
```

Comment:
Since the type of the result register is known, this could be condensed to be a bit simpler, making use of the `i32` and `i64` constructors from `TypedReg`:

```rust 
use OperandSize::*;
match size {
  S32 => self.stack.push(TypedReg::i32(dst).into()),
  S64 => self.stack.push(TypedReg::i64(dst).into()),
  _ unreachable!()
}
```